src/ophyd_async/testing/_assert.py

```diff
@@ -18,6 +24,20 @@ def _generate_assert_error_msg(name: str, expected_result, actual_result) -> str
     )
 
 
+def _value_equal(value1, value2) -> bool:
+    return value1 == pytest.approx(value2)
+
+
+def _table_equal(table1: Table, table2: Table) -> bool:
+    # keys and values of all fields should be the same
+    if not table1.model_fields == table2.model_fields:
+        return False
+    for field, value1 in table1:
+        if not _value_equal(value1, getattr(table2, field)):
+            return False
+    return True
```

Comment:
I've got another strategy for you:
```python
class EqualSignalValue:
    def __init__(self, value: SignalDatatype):
        self.value = value
    def __ne__(self, other):
        return _is_different(other, self.value)  # this function should be extracted from ophyd_async.plan_stubs._settings::apply_settings, renamed, and put in core

async def assert_reading(readable: AsyncReadable, expected_reading: dict[str, Reading]):
    actual_reading = await readable.read()
    expected_reading = {k: dict(**v, value=EqualSignalValue(v["value"])) for k, v in expected_reading.items()}
    assert actual_reading == expected_reading
```
Then you don't need to do the waking in each function, just change each value into an `EqualSignalValue`. You might need to read up on which of `__eq__`, `__ne__` you need to override to make this work properly